### PR TITLE
Run JSHint on Test Suite

### PR DESCRIPTION
**Pull request by seanpdoyle**
Test suite should comply with JSHint standards and conventions

### CODE REVIEW

File: packages/ember-data/tests/unit/transform/number_test.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ test("#deserialize", function() {
   equal(transform.deserialize(undefined),       null);
   equal(transform.deserialize("1.1"),           1.1);
   equal(transform.deserialize(1.1),             1.1);
-  equal(transform.deserialize(new Number(1.1)), 1.1);
+  equal(transform.deserialize(Number(1.1)), 1.1);
```
Comment (fivetanley) [inline packages/ember-data/tests/unit/transform/number_test.js]: These ones we want to stay the same as they are testing Number objects (not primitives) are being deserialized correctly.

Comment (seanpdoyle) [inline packages/ember-data/tests/unit/transform/number_test.js]: This change was suggested by JSHint.

http://stackoverflow.com/questions/369220/why-should-you-not-use-number-as-a-constructor

Should I turn off the rule globally?

Comment (igorT) [inline packages/ember-data/tests/unit/transform/number_test.js]: It's a good global rule, but I think that here we are specifically testing for people who passed in a number object.

File: packages/ember-data/tests/unit/transform/number_test.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+/* jshint -W053 */
+
```
Comment (seanpdoyle) [inline packages/ember-data/tests/unit/transform/number_test.js]: @fivetanley @igorT :+1:

**Comment by bmac** [comment]
Thanks @seanpdoyle!

**Pull request merged**